src/types.jl

```diff
@@ -118,6 +118,38 @@ function Base.show(io::IO, key::PublicKey)
     print("PublicKey{", backend_name(key), "}()")
 end
 
+mutable struct SecureMatrix{CryptoBackendT <: AbstractCryptoBackend, DataT}
```

Comment:
Maybe add a short comment what capacity means in this context (also to `PlainMatrix` below)? E.g., something like
```suggestion
# Here `capacity` needs to be large enough to hold at least `prod(shape)` elements
mutable struct SecureMatrix{CryptoBackendT <: AbstractCryptoBackend, DataT}
```
?